src/diffusers/models/transformers/transformer_wan_s2v.py

```diff
@@ -0,0 +1,1218 @@
+    _no_split_modules = ["WanS2VTransformerBlock"]
+    _keep_in_fp32_modules = ["time_embedder", "scale_shift_table", "norm1", "norm2", "norm3", "causal_audio_encoder"]
+    _keys_to_ignore_on_load_unexpected = ["norm_added_q"]
+    _repeated_blocks = ["WanS2VTransformerBlock"]
+
+    @register_to_config
+    def __init__(
+        self,
+        patch_size: Tuple[int] = (1, 2, 2),
+        num_attention_heads: int = 40,
+        attention_head_dim: int = 128,
+        in_channels: int = 16,
+        out_channels: int = 16,
+        text_dim: int = 4096,
+        freq_dim: int = 256,
+        audio_dim: int = 1024,
+        audio_inject_layers: List[int] = [0, 4, 8, 12, 16, 20, 24, 27, 30, 33, 36, 39],
+        enable_adain: bool = True,
+        adain_mode: str = "attn_norm",
+        pose_dim: int = 16,
+        ffn_dim: int = 13824,
+        num_layers: int = 40,
+        cross_attn_norm: bool = True,
+        qk_norm: Optional[str] = "rms_norm_across_heads",
+        eps: float = 1e-6,
+        added_kv_proj_dim: Optional[int] = None,
+        rope_max_seq_len: int = 1024,
+        enable_framepack: bool = True,
+        framepack_drop_mode: str = "padd",
+        add_last_motion: bool = True,
+        zero_timestep: bool = True,
+    ) -> None:
+        super().__init__()
+
+        self.inner_dim = inner_dim = num_attention_heads * attention_head_dim
+        out_channels = out_channels or in_channels
+
+        # 1. Patch & position embedding
+        self.rope = WanS2VRotaryPosEmbed(attention_head_dim, patch_size, rope_max_seq_len, num_attention_heads)
+        self.patch_embedding = nn.Conv3d(in_channels, inner_dim, kernel_size=patch_size, stride=patch_size)
+
+        if enable_framepack:
+            self.frame_packer = FramePackMotioner(
+                inner_dim=inner_dim,
+                num_attention_heads=num_attention_heads,
+                zip_frame_buckets=[1, 2, 16],
+                drop_mode=framepack_drop_mode,
+                patch_size=patch_size,
+            )
+
+        self.trainable_condition_mask = nn.Embedding(3, inner_dim)
+
+        # 2. Condition Embeddings
+        self.condition_embedder = WanTimeTextAudioPoseEmbedding(
+            dim=inner_dim,
+            time_freq_dim=freq_dim,
+            time_proj_dim=inner_dim * 6,
+            text_embed_dim=text_dim,
+            audio_embed_dim=audio_dim,
+            pose_embed_dim=pose_dim,
+            patch_size=patch_size,
+            enable_adain=enable_adain,
+        )
+
+        # 3. Transformer blocks
+        self.blocks = nn.ModuleList(
+            [
+                WanS2VTransformerBlock(
+                    inner_dim, ffn_dim, num_attention_heads, qk_norm, cross_attn_norm, eps, added_kv_proj_dim
+                )
+                for _ in range(num_layers)
+            ]
+        )
+
+        # 4. Audio Injector
+        all_modules, all_modules_names = torch_dfs(self.blocks, parent_name="root.transformer_blocks")
```

Comment:
let's not make this so complex, if you have the transformer's configuration, we should be able to config the AudioInjection - no need to dynamically search for tranformer_blocks